py3status/modules/systemd.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""
+Check systemd unit status.
+
+Check the status of a systemd unit.
+Color will be GOOD if is active.
+BAD if inactive.
+DEGRADED if unit could not be found.
+
+Configuration parameters:
+    cache_timeout: How often we refresh this module in seconds (default 5)
+    format: Format for module output (default "{unit}: {status}")
+    unit: Name of the unit (default None)
+
+Format of status string placeholders:
+    {unit} name of the unit
+    {status} 'active', 'inactive' or 'not-found'
+
+Color options:
+    color_good: Unit active
+    color_bad: Unit inactive
+    color_degraded: Unit not found
+
+Example:
+
+```
+# Check status of vpn service
+# Start with left click
+# Stop with right click
+systemd vpn {
+    unit = 'vpn.service'
+    on_click 1 = "exec sudo systemctl start vpn"
+    on_click 3 = "exec sudo systemctl stop vpn"
+    format = '🏢'
+}
+```
+
+Requires:
+    pydbus: python lib for dbus
+
+@author Adrian Lopez <adrianlzt@gmail.com>
+@license BSD
+"""
+
+from time import time
+from pydbus import SystemBus
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 5
+    format = '{unit}: {status}'
+    unit = None
+
+    def __init__(self):
+        """
+        This is the class constructor which will be executed once.
+        """
+        self.systemd_unit = None
+
+    def post_config_hook(self):
+        bus = SystemBus()
+        systemd = bus.get('org.freedesktop.systemd1')
+        s_unit = systemd.LoadUnit(self.unit)
+        self.systemd_unit = bus.get('.systemd1', s_unit)
+
+    def check_status(self, i3s_output_list, i3s_config):
+        """
+        Ask dbus to get Status and loaded status for the unit
+        """
+        status = self.systemd_unit.Get('org.freedesktop.systemd1.Unit', 'ActiveState')
+        exists = self.systemd_unit.Get('org.freedesktop.systemd1.Unit', 'LoadState')
+
+        if exists == 'not-found':
+            color = self.py3.COLOR_DEGRADED
+            status = exists
+        elif status == 'active':
+            color = self.py3.COLOR_GOOD
+        elif status == 'inactive':
+            color = self.py3.COLOR_BAD
+        else:
+            color = self.py3.COLOR_DEGRADED
+
+        full_text = self.format.format(unit=self.unit, status=status)
```

Comment:
I take the [example](https://github.com/ultrabug/py3status/blob/master/doc/example_module.py#L116) module as base. Maybe should be fixed.